vsts/determine_tests_to_run.ps1

```diff
@@ -0,0 +1,92 @@
+﻿# For pull requests, this script checks the git diff between the merging branch and the upstream branch to see
+#  if any code changes were made to iothub clients or provisioning clients. As a result, this script sets environment
+#  variables to signal later vsts scripts to run or not to run certain e2e tests
+
+# For all other builds (nightly builds, for instance), this script will set environment variables to run all e2e tests.
+
+Write-Host "Determining tests to run..."
+$Env:RUN_IOTHUB_TESTS = "false"
+$Env:RUN_PROVISIONING_TESTS = "false"
+
+$targetBranch = ($env:TARGET_BRANCH)
+if (($env:TARGET_BRANCH).toLower().Contains("system.pullrequest.targetbranch"))
```

Comment:
Question: what is stored in `$env:TARGET_BRANCH`?

Reply:
It is equal to the azure devops built-in variable $(System.PullRequest.TargetBranch)

Its value is the target branch of the PR build ("master", usually) or it is a default value for builds that are not PR builds. I'll add a note in the script about this